src/mono/mono/mini/aot-compiler.c

```diff
@@ -6111,6 +6114,90 @@ method_is_externally_callable (MonoAotCompile *acfg, MonoMethod *method)
 	}
 }
 
+#ifdef MONO_ARCH_AOT_SUPPORTED
+static gboolean
+method_to_pinvoke_has_scope_import (MonoAotCompile *acfg, MonoMethod *method, const char **module, const char **entrypoint)
+{
+	MonoImage *image = m_class_get_image (method->klass);
+	MonoMethodPInvoke *piinfo = (MonoMethodPInvoke *) method;
+	MonoTableInfo *tables = image->tables;
+	MonoTableInfo *im = &tables [MONO_TABLE_IMPLMAP];
+	MonoTableInfo *mr = &tables [MONO_TABLE_MODULEREF];
+	guint32 im_cols [MONO_IMPLMAP_SIZE];
+	int module_idx;
+	const char **scope_import;
+	guint32 scope_token;
+
+	if (g_hash_table_lookup_extended (acfg->method_to_pinvoke_scope_import, method, NULL, (gpointer *)&scope_import) && scope_import) {
+		if (module)
+			*module = g_strdup (scope_import[0]);
```

Comment:
Ownership of these strings are in the hash table, and returned as const char ** meaning caller can use them but won't need to free them, so non need to duplicate the strings.

Reply:
Thanks for pointing this out! I had been setting them to `g_strdup (string val)` without thinking twice, and made my self think I needed some way to free the allocated memory in the function caller.